packages/agents/lighthouse/src/tasks/sendOutboundRoot/helpers/bnb.ts

```diff
@@ -33,14 +33,20 @@ export const getSendOutboundRootParams = async (l2domain: string): Promise<Extra
     throw new NoSpokeConnector(l2ChainId, requestContext, methodContext);
   }
 
-  let encodedData = contracts.spokeConnector.encodeFunctionData("AMB");
-  let encoded = await chainreader.readTx({ data: encodedData, domain: Number(l2domain), to: l2SpokeConnector.address });
-  const [ambAddress] = contracts.spokeConnector.decodeFunctionResult("AMB", encoded);
+  const l2Provider = getJsonRpcProvider(l2RpcUrl);
+  const l2SpokeConnectorContract = getContract(
+    l2SpokeConnector.address,
+    l2SpokeConnector.abi as ContractInterface,
+    l2Provider,
+  );
+  const ambAddress = await l2SpokeConnectorContract.AMB();
 
-  const ambInterface = getInterface(ambs.bnb);
-  encodedData = ambInterface.encodeFunctionData("calcSrcFees", ["", hubChainId, 32]);
-  encoded = await chainreader.readTx({ data: encodedData, domain: Number(l2domain), to: ambAddress });
-  const [_fee] = ambInterface.decodeFunctionResult("calcSrcFees", encoded);
+  // gasLimit on hub side = 200_000
+  // actually it required about 100_000 gas on mainnet to call `receiveWormholeMessages`
+  // remain gas will be refunded on mainnetto refund address (default: deployer address)
+  const gasLimit = "200000";
+  const fee = await l2SpokeConnectorContract.quoteEVMDeliveryPrice(gasLimit, ambAddress);
```

Comment:
why are we using the rpc directly instead of the chain reader?

Reply:
We used `rpc` from `getBestProvider` function, so i thought that would be no problem